src/ipc/ipc4/logging.c

```diff
@@ -150,27 +154,47 @@ int ipc4_logging_enable_logs(bool first_block,
 	return 0;
 }
 
-int ipc4_logging_shutdown(void)
-{
-	struct ipc4_log_state_info log_state = { 0 };
-
-	/* log_state.enable set to 0 above */
-
-	return ipc4_logging_enable_logs(true, true, sizeof(log_state), (char *)&log_state);
-}
+#endif
 
-#else
+#ifdef CONFIG_LOG_BACKEND_SOF_PROBE
 int ipc4_logging_enable_logs(bool first_block,
 			     bool last_block,
 			     uint32_t data_offset_or_size,
 			     const char *data)
 {
-	return IPC4_UNKNOWN_MESSAGE_TYPE;
-}
+	const struct log_backend *log_backend = log_backend_probe_get();
+	const struct ipc4_log_state_info *log_state;
+
+	if (!(first_block && last_block))
+		return -EINVAL;
+
+	if (data_offset_or_size < sizeof(struct ipc4_log_state_info))
+		return -EINVAL;
+
+	dcache_invalidate_region((__sparse_force void __sparse_cache *)data, data_offset_or_size);
+
+	log_state = (const struct ipc4_log_state_info *)data;
+
+	if (log_state->enable) {
+		probe_logging_init(probe_logging_hook);
```

Comment:
Hmm, I think we have a bit of a conlict. In current code, probe loggign is enabled by adding a probe point for the logs:

sof/src/probe/probe.c:probe_point_add()

@abonislawski added this (see git history).

So not sure how probe add and IPC4 log enable should relate to each other. I guess we could support both ways to enable probe log. Need to investigate more how this works.